tests/countries/test_micronesia.py

```diff
@@ -0,0 +1,538 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from unittest import TestCase
+
+from holidays.countries.micronesia import Micronesia, FM, FSM
+from tests.common import CommonCountryTests
+
+
+class TestMicronesia(CommonCountryTests, TestCase):
+    @classmethod
+    def setUpClass(cls):
+        years = range(1987, 2050)
+        super().setUpClass(Micronesia, years=years, years_non_observed=years)
+        cls.subdiv_holidays = {
+            subdiv: FM(subdiv=subdiv, years=years) for subdiv in FM.subdivisions
+        }
+
+    def test_country_aliases(self):
+        self.assertAliases(Micronesia, FM, FSM)
+
+    def test_no_holidays(self):
+        self.assertNoHolidays(Micronesia(years=1986))
+
+    def test_new_years_day(self):
+        name = "New Year's Day"
+        self.assertHolidayName(name, (f"{year}-01-01" for year in range(1987, 2050)))
+        dt = (
+            "2006-01-02",
+            "2012-01-02",
+            "2017-01-02",
+            "2023-01-02",
+        )
+        self.assertHolidayName(f"{name} (observed)", dt)
+        self.assertNoNonObservedHoliday(dt)
+
+    def test_micronesian_culture_and_tradition_day(self):
+        name = "Micronesian Culture and Tradition Day"
+        self.assertHolidayName(name, (f"{year}-03-31" for year in range(2011, 2050)))
```

Comment:
```suggestion
        self.assertHolidayName(name, (f"{year}-03-31" for year in range(2011, 2050)))
        self.assertNoHolidayName(name, range(1987, 2011))
```
And same for all holidays that are introduced later.